src/Generator/Program.fs

```diff
@@ -0,0 +1,390 @@
+        match ty with
+        | Pointer ty -> unwrapTyFromPointer ty
+        | _ -> ty
+    let transformPointerTy i transform ty =
+        let transformPointerTy ty =
+            match ty with
+            | Pointer(inner) ->
+                let flattenedTy = unwrapTyFromPointer inner
+                let res = flattenedTy |> transform
+                if res = flattenedTy then ty
+                else res
+            | inner -> inner
+        match ty with
+        | Pointer(GLchar) ->
+            None, GLString
+        | Pointer(Pointer(GLchar)) ->
+            None, GLString |> ArrayType
+        | Pointer(_) when unwrapTyFromPointer ty = Void ->
+            let name = 
+                ("TElement" + string i)
+            let inner = name |> StructGenericType
+            let res = inner |> transform 
+            if res = inner then None, ty
+            else Some name, res
+        | _ -> None, transformPointerTy ty
+    if lengthParamsSet.Count = 0 then
+        let adjustedParameters =
+            func.parameters
+            |> Array.map(fun currParameter ->
+                match currParameter.typ.typ with
+                | Pointer(GLchar) ->
+                    { currParameter with
+                        PrintReadyTypedParameterInfo.typ = GLString |> PrintReady.formatTypeInfo }
+                | Pointer(Pointer(GLchar)) ->
+                    { currParameter with
+                        PrintReadyTypedParameterInfo.typ = GLString |> ArrayType |> PrintReady.formatTypeInfo }
+                | _ -> currParameter)
+        { func with parameters = adjustedParameters } |> Array.singleton
+    else
+        let overloadWithMapping tyMapper =
+            let mutable i = 1
+            let adjustedParameters =
+                func.parameters
+                |> Array.map(fun currParameter ->
+                    if lengthParamsSet.Contains currParameter.actualName then
+                        let _, newParameterType =
+                            currParameter.typ.typ
+                            |> transformPointerTy 0 RefPointer
+                        None, { currParameter with typ = newParameterType |> PrintReady.formatTypeInfo }
+                    else
+                        let (genericName, newParameterType) =
+                            currParameter.typ.typ 
+                            |> transformPointerTy i tyMapper
+                        genericName |> Option.iter(fun _ -> i <- i + 1)
+                        genericName, { currParameter with typ = newParameterType |> PrintReady.formatTypeInfo })
+            let genericTypes = adjustedParameters |> Array.choose fst
+            let parameters = adjustedParameters |> Array.map snd
+            { func with
+                genericTypes = genericTypes
+                parameters = parameters }
+        pointerTypeMappings
+        |> Array.Parallel.map overloadWithMapping
+
+[<RequireQualifiedAccess>]
+type PointerCanidate =
```

Comment:
Is this a type used during debugging purposes or is it going to be used in production?

Reply:
Well it was a pun and I thought I removed it. Apparently I did oversee it. Funny enough to have another person to see it once